### PR TITLE
po/deperacte clipping

### PR DESCRIPTION
**Pull request by TurboGit**
- deprecate clipping
- update presets
- update quick access panel

Let's get ready for 3.8 !

### CODE REVIEW

File: src/libs/modulegroups.c [hunk 1]
```diff
@@ -1830,9 +1825,11 @@ void init_presets(dt_lib_module_t *self)
   AM("tonemap");
   AM("vibrance");
   AM("basicadj");
-  // this modules are deprecated in 3.6 and should be removed 1 yer later
+  // these modules are deprecated in 3.6 and should be removed 1 year later
   AM("spots");
   AM("defringe");
+  // these modules are deprecated in 3.8 and should be removed 1 year later
+  AM("clipping");
```
Comment (AlicVB) [inline src/libs/modulegroups.c:1832]: do we want to take the opportunity to "remove" 3.4 deprecated modules too ?
Or we can do it later...

File: src/libs/modulegroups.c [hunk 2]
```diff
@@ -1523,7 +1523,6 @@ void init_presets(dt_lib_module_t *self)
 
   SMG(C_("modulegroup", "base"), "basic");
   AM("basecurve");
-  AM("clipping");
```
Comment (AlicVB) [inline src/libs/modulegroups.c:1526]: I would have put "ashift" here (and remove it from it's actual group) as it now handle rotation which belong to the basic group imho (same for other presets)

File: src/iop/clipping.c [hunk 3]
```diff
@@ -299,6 +299,10 @@ static inline void adjust_aabb(const float *p, float *aabb)
   aabb[3] = fmaxf(aabb[3], p[1]);
 }
 
+const char *deprecated_msg()
+{
+  return _("this module is deprecated. please use the crop and/or perspective correction modules instead.");
```
Comment (AlicVB) [inline src/iop/clipping.c:304]: I agree with that, but I prefer to raise 2 comments to be sure : 
- there's also the orientation module for flipping. but it's not a key feature, so we may prefer to forget it and keep the msg simpler
- do we want to include here a warning about double crop or is it self explanatory ?

File: src/libs/modulegroups.c (hunk 1 above)
Comment (TurboGit) [reply src/libs/modulegroups.c:1832]: Later, in another PR as this will certainly get some discussions :)

**Comment by TurboGit** [comment]
@AlicVB : Fixed version just pushed.

**Comment by AlicVB** [comment]
@elstoc : for the docs, seems we don't have anything about what is a deprecated module. Considering the number of questions we already have (and in this specific case we certainly will have a lot more) it would be awesome to have something in the docs.
You know my English level, so it would be better if someone else than me write it, but I can write a draft if needed...

**Comment by elstoc** [comment]
@AlicVB we have a fairly standard approach for deprecated modules. See [here](https://darktable-org.github.io/dtdocs/module-reference/processing-modules/channel-mixer/) for example. I'm happy to pick up the documentation work - there are a few outstanding changes to add now. I'll work through the documentation-pending PRs in the next few weeks.

File: src/iop/clipping.c (hunk 3 above)
Comment (TurboGit) [reply src/iop/clipping.c:304]: Good catch ! Now fixed thanks.

**Comment by elstoc** [comment]
Ok so a new section about deprecated modules in general. Perhaps even separating the deprecated modules into their own sub-section. I'll raise an issue against the dtdocs repo to remind me and consider what the best approach is.

**Pull request merged**